datadog/dogstatsd/base.py

```diff
@@ -418,16 +428,17 @@ def _is_telemetry_flush_time(self):
         return False
 
     def _send_to_server(self, packet):
-        flush_telemetry = self._is_telemetry_flush_time()
-        if flush_telemetry:
-            packet += "\n"+self._flush_telemetry()
+        self._xmit_packet(packet)
+        if self._is_telemetry_flush_time():
+            telemetry = self._flush_telemetry()
+            self._reset_telemetry()
+            self._xmit_packet(telemetry)
```

Comment:
We're losing telemetry info here in case we fail to send the telemetry over the network since we reset it in any case. `_xmit_packet` should return a value to indicate if it was successful or not.

Reply:
Indeed I will adjust to keep telemetry if a failure occur while sending the packet.